test/e2e-common.sh

```diff
@@ -137,11 +137,19 @@ function test_task_creation() {
         done
 
         local cnt=0
+        local all_status=''
+        local reson=''
         while true;do
             [[ ${cnt} == ${maxloop} ]] && show_failure ${testname} ${tns}
 
-            all_status=$(kubectl get -n ${tns} pipelinerun --output=jsonpath='{.items[*].status.conditions[*].status}')
-            reason=$(kubectl get -n ${tns} pipelinerun --output=jsonpath='{.items[*].status.conditions[*].reason}')
+            # sometimes we don't get all_status and reason in one go so
+            # wait until we get the reason and all_status for 5 iterations
+            for _ in {1..5}; do
+                all_status=$(kubectl get -n ${tns} pipelinerun --output=jsonpath='{.items[*].status.conditions[*].status}')
+                reason=$(kubectl get -n ${tns} pipelinerun --output=jsonpath='{.items[*].status.conditions[*].reason}')
+                [[ ! -z ${all_status} ]] && [[ ! -z ${reason} ]] && break
+            done
+            
 
             if [[ -z ${all_status} && -z ${reason} ]];then
```

Comment:
we can remove this then

Reply:
we need line no 154 as well because here we are checking for `taskrun` and earlier we checked for `pipelinerun` and in tests we can have anything be it a `taskrun` or a `pipelinerun`

Reply:
should we do the same as above then here also

Reply:
:thinking: by that time we will get the status and reason both. Actually when the while loop starts the CI tries to look for the status and reason immediately and therefore for the first time it doesn't gets anything

Reply:
addressed